src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/editdocs/EditDocsAuditHearingRecordingDocTypeService.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.postsubmit.editdocs;
+
+import static uk.gov.hmcts.reform.iacaseapi.domain.handlers.postsubmit.editdocs.EditDocsAuditMetadataTypeService.getIdFromDocUrl;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.HearingRecordingDocument;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.IdValue;
+
+@Component
+@Qualifier("hearingRecordingDocAudit")
+public class EditDocsAuditHearingRecordingDocTypeService implements EditDocsAuditService {
+    @Override
+    public List<String> getUpdatedAndDeletedDocIdsForGivenField(AsylumCase asylumCase, AsylumCase asylumCaseBefore,
+                                                                AsylumCaseFieldDefinition field) {
+        List<IdValue<HearingRecordingDocument>> doc = getDocField(asylumCase, field);
```

Comment:
`EditDocsAuditHearingRecordingDocTypeService` and `EditDocsAuditMetadataTypeService` are very similar classes,
I am just wondering if you add
```
interface HasDocument {
   Document getDocument();
}
```
to `HearingRecordingDocument` and `DocumentWithMetadata` it could be reduced to one service.

Reply:
yes, agree, however, I have to admit couldn't find an easy or nice way to do it. The issues I have is that they both are similar but they have to use a different caseDefinitionField and then there is not a nice way to infer the type at running time. I can probably do it using instance of or similar but as I said did not like it so I left it this way in the end. But happy to look into this if I have some more time or if you have any ideas. Thanks

Reply:
this is now addressed. Thanks a lot for suggesting this solution once again! It simplifies the code a lot by reusing it.